device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -54,10 +54,10 @@ inline void aggregate_cluster(
     scalar totalWeight = 0.;
     point2 mean{0., 0.}, var{0., 0.}, offset{0., 0.};
 
-    const auto module_link = cells[cid + start].module_link;
+    const unsigned int module_idx = cells.module_index().at(cid + start);
     const unsigned short partition_size = end - start;
-    const scalar pitch_x = det_descr.pitch_x().at(module_link);
-    const scalar pitch_y = det_descr.pitch_y().at(module_link);
+    const scalar pitch_x = det_descr.pitch_x().at(module_idx);
+    const scalar pitch_y = det_descr.pitch_y().at(module_idx);
```

Comment:
```suggestion
    const scalar pitch_x = det_descr.at(module_idx).pitch_x();
    const scalar pitch_y = det_descr.at(module_idx).pitch_y();
```

Reply:
Here I could be convinced. :thinking: Though at that point I'd probably write:

```suggestion
    const auto module_descr = det_descr.at(module_idx);
    const scalar pitch_x = module_descr.pitch_x();
    const scalar pitch_y = module_descr.pitch_y();
```

Sure, this I can do...

Reply:
So... mea culpa. Indeed, in this function the same cell is taken for practically all of its info. So now I switched (back) to accessing a proxy for a given cell, and then using that proxy pretty much everywhere. On top of doing the "`module_descr` change" that I teased earlier.